tests/test_ble_interface_core.py

```
@@ -4947,8 +5090,426 @@ def _record_noop_start(thread: object) -> None:
         assert start_calls == [thread_like]
-        assert iface._receiveThread is None
+        assert iface._receiveThread is thread_like
     finally:
+        with iface._state_lock:
+            iface._receiveThread = original_receive_thread
+            iface._receive_start_pending = original_receive_start_pending
+            iface._receive_start_pending_since = original_receive_start_pending_since
+        iface.close()
+
+
+def test_start_receive_thread_pending_marker_allows_later_restart(
```

Ruff (0.15.6)

[warning] Too many statements (52 > 50)

(PLR0915)